bindings/rust/build.rs

```diff
@@ -69,6 +69,9 @@ fn build_with_cmake() {
     let compiler = cc::Build::new().get_compiler();
     let has_ninja = if compiler.is_like_msvc() {
         setup_env_msvc(&compiler);
+        // this is a BIG HACK that should be fixed in unicorn's cmake!!
```

Comment:
I needed this for `cargo test` to succeed on my machine...the linking stuff in the CMakeLists.txt is pretty messy, but someone else can fix it..

Reply:
Weird, what symbol collisions do you get?

Reply:
here is how `target\debug\build\unicorn-engine-<hash>\output` looks after e.g. `cargo clean && cargo build`:
https://gist.github.com/shuffle2/f9623c334f1ea82b3d995c5ef6155d0a

when you run `cargo test`, it links an executable instead of a library, so these `LNK4006` warnings turn into errors.

Reply:
ah yea, looks like ubuntu CI builder had same issue

Reply:
for now i've added the equivalent flag to non-msvc build

Reply:
ugh, let me fix + test locally on linux first

Reply:
should be fixed now, will have to see if it also works for clang/macos

Reply:
sigh, so lld on macos doesn't support the flag...

Reply:
hm, did it not cause an error there to begin with? 🤔 https://github.com/unicorn-engine/unicorn/runs/6035089850?check_suite_focus=true

Reply:
Seems so. I would also have a try locally tonight.

Reply:
do you know what `cc` (in PATH) and `CC` (env var) are set to on mac os?

Reply:
> do you know what `cc` (in PATH) and `CC` (env var) are set to on mac os? It's obviously clang, but it seems to screw up the detection the rust cc crate is using to identify the compiler
> 
> probably something like this: [alexcrichton/cc-rs#408](https://github.com/alexcrichton/cc-rs/issues/408)

cc is AppleClang on recent macOS but cmake might wrongly detect it

Reply:
yea it seems cc is an actual binary (not a symlink). in any case i've worked around it